db/solr-ec2.yml

```diff
@@ -0,0 +1,114 @@
+# db/solr-ec2.yml
+AWSTemplateFormatVersion: "2010-09-09"
+Parameters:
+  KeyPairName:
+    Description: >
+      The EC2 key pair used for the instance
+    Type: AWS::EC2::KeyPair::KeyName
+  EnvironmentType:
+    Type: String
+    Description: Environment this instance is used by
+    Default: Staging
+    AllowedValues:
+      - Staging
+      - Production
+  VPCStackName:
+    Type: String
+    Description: VPC stack to launch the instance into
+  ImageId:
+    Type: AWS::EC2::Image::Id
+  AlarmSnsTopic:
+    Type: String
+    Description: Optional topic arn for CloudWatch alarms
+  UniqueName:
+    Type: String
+    Description: Unique name for associated resources (CamelCase; should not include env name)
+    AllowedPattern: ^[a-zA-Z][a-zA-Z0-9\-]*$
+  InstanceType:
+    Type: String
+    AllowedValues:
+      - t3.nano
+      - t3.micro
+      - t3.small
+      - t3.medium
+      - t3.large
+  DataVolumeSize:
+    Type: Number
+Conditions:
+  IsProduction: !Equals [!Ref EnvironmentType, Production]
+  CreateAlarms: !Not [!Equals [!Ref AlarmSnsTopic, ""]]
+Resources:
+  Ec2Instance:
+    Type: AWS::EC2::Instance
+    Properties:
+      InstanceType: !Ref InstanceType
+      ImageId: !Ref ImageId
+      KeyName: !Ref KeyPairName
+      SecurityGroupIds:
+        - !GetAtt SecurityGroup.GroupId
+      SubnetId:
+        Fn::ImportValue:
+          !Sub ${VPCStackName}-Subnet1
+      Tags:
+        - Key: Project
+          Value: Solr
+        - Key: Environment
+          Value: !Ref EnvironmentType
+        - Key: "prx:cloudformation:stack-name"
+          Value: !Ref AWS::StackName
+        - Key: "prx:cloudformation:stack-id"
+          Value: !Ref AWS::StackId
+  SolrDataEbsVolume:
+    Type: AWS::EC2::Volume
+    DeletionPolicy: Retain
+    UpdateReplacePolicy: Retain
+    Properties:
+      AvailabilityZone: !GetAtt Ec2Instance.AvailabilityZone
+      Size: !Ref DataVolumeSize
+      Tags:
+        - Key: Project
+          Value: Solr
+        - Key: Environment
+          Value: !Ref EnvironmentType
+        - Key: "prx:cloudformation:stack-name"
+          Value: !Ref AWS::StackName
+        - Key: "prx:cloudformation:stack-id"
+          Value: !Ref AWS::StackId
+  SolrDataEbsVolumeAttachent:
+    Type: AWS::EC2::VolumeAttachment
+    Properties:
+      Device: /dev/xvdb
```

Comment:
this is not the name I see for the device.
the ebs volumes are mounted on:
```
/dev/nvme0n1p1  7.8G  1.4G  6.3G  18% /
/dev/nvme1n1    7.8G  766M  6.6G  11% /data
```

Reply:
This seems to be the expected behavior, so nothing much to be done in Cfn here:
`The device names that you specify in a block device mapping are renamed using NVMe device names (/dev/nvme[0-26]n1).`
https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/nvme-ebs-volumes.html